src/lang/queryAst.ts

```diff
@@ -518,8 +525,15 @@ export function isLinesParallelAndConstrained(
     if (err(_primarySegment)) return _primarySegment
     const primarySegment = _primarySegment.segment
 
+    const _varDec2 = getNodeFromPath(ast, secondaryPath, 'VariableDeclaration')
+    if (err(_varDec2)) return _varDec2
+    const varDec2 = _varDec2.node
+    const varName2 = (varDec2 as VariableDeclaration)?.declaration.id?.name
+    const sg2 = sketchFromKclValue(memVars[varName2], varName2)
+    if (err(sg2)) return sg2
+
     const _segment = getSketchSegmentFromSourceRange(
-      sg,
+      sg2,
       secondaryLine?.codeRef?.range
```

Comment:
Is this making it so that this query handles if the segments are in different sketch profiles?

Reply:
The constraints, right.
I can't remember how far I got with these I might have started to get them to work across profiles, but not sure where I got up.

I think that's likely what's happening here.

Reply:
Yeah okay I recall you saying way back that you were dabbling in that, I think that can be future work. Would love to alert users that it's not possible with a toast if they try it, but that can be a follow-up while multi-profile is nightly or something?